mini-projects/00-jq-rs/src/main.rs

```diff
@@ -1,3 +1,112 @@
-fn main() {
-    println!("Hello World!");
+use anyhow::{Context, Result};
+use clap::Parser;
+use serde_json::Value;
+use std::{
+    env,
+    io::{stdout, BufReader},
+    num::ParseIntError,
+    path::PathBuf,
+};
+use thiserror::Error;
+
+mod filters;
+use filters::pipe;
+
+mod functions;
+mod output;
+use output::print_result;
+
+#[derive(Error, Debug)]
+enum MyErrors {
+    #[error("The indent value must be in the range of 0-7 inclusive. Current value: {0}")]
+    InvalidIndent(usize),
+
+    #[error("You can have either colored or monochrome output")]
+    ColoredAndMonochromeError,
+
+    #[error("You can have either compact or indented output")]
+    CompactAndIndentedError,
```

Comment:
nit: These probably could have been combined into a `ArgParsingError`, where the actual error message can be a parameter. This would make adding in new conflicting argument cases easier